homeassistant/components/tedee/coordinator.py

```diff
@@ -0,0 +1,103 @@
+"""Coordinator for Tedee locks."""
+import asyncio
+from datetime import timedelta
+import logging
+import time
+
+from pytedee_async import (
+    TedeeClient,
+    TedeeClientException,
+    TedeeDataUpdateException,
+    TedeeLocalAuthException,
+    TedeeLock,
+)
+
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_LOCAL_ACCESS_TOKEN, DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=20)
+STALE_DATA_INTERVAL = 300
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TedeeApiCoordinator(DataUpdateCoordinator[dict[int, TedeeLock]]):
+    """Class to handle fetching data from the tedee API centrally."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+        if not self.config_entry:
+            raise ConfigEntryError("Config entry not set")
```

Comment:
Oh right, please overwrite the type to `config_entry: ConfigEntry`, checkout YouTube coordinator for example